File: examples/python/update_memory.py
import json
import sys
from urllib.parse import quote

from memside_http import api_request


if len(sys.argv) < 4:
    raise RuntimeError(
        "Usage: python update_memory.py "
        "<memory-id> <expected-version> <new-text>"
    )

memory_id = sys.argv[1]
expected_version = int(sys.argv[2])
text = " ".join(sys.argv[3:]).strip()
updated = api_request(
    "PATCH",
    f"/memories/{quote(memory_id, safe='')}",
    body={"text": text, "expected_version": expected_version},
)

print(json.dumps(updated, indent=2))

File: examples/javascript/update-memory.mjs
import { memside } from "./memside-client.mjs";

const [memoryId, expectedVersionText, ...textParts] = process.argv.slice(2);
const expectedVersion = Number(expectedVersionText);
const text = textParts.join(" ").trim();

if (!memoryId || !Number.isInteger(expectedVersion) || !text) {
  throw new Error(
    "Usage: node update-memory.mjs <memory-id> <expected-version> <new-text>"
  );
}

const updated = await memside(
  `/memories/${encodeURIComponent(memoryId)}`,
  {
    method: "PATCH",
    body: JSON.stringify({
      text,
      expected_version: expectedVersion,
    }),
  }
);

console.log(JSON.stringify(updated, null, 2));
